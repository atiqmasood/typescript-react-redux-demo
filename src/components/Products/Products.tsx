import * as React from "react";

import { Card, Grid, Image } from "semantic-ui-react";
import { IProduct, IProductProps, IProps } from "../../typings/products";
/**
 * Single Product view
 */
const SingleProduct = (props: IProductProps) => {
  const { product } = props;
  return (
    <Card key={product.id + "card"}>
      <Image
        src={`//cdn.aboutstatic.com/file/${
          product.image.hash
        }?quality=90&progressive=1&bg=f2f2f2&brightness=0.95&width=400&height=400`}
      />
      <Card.Content>
        <Card.Header>{product.name}</Card.Header>
        <Card.Meta>
          Price : {product.min_price} - {product.max_price}
        </Card.Meta>
        <Card.Description>{product.brand.name}</Card.Description>
      </Card.Content>
      <Card.Content extra={true}>
        Quantity: {product.quantity_per_pack}
      </Card.Content>
    </Card>
  );
};
/**
 * Products lists iterated to display single products
 */
const Products = (props: IProps) => {
  const { products } = props;
  return (
    <React.Fragment>
      <Grid columns={4} divided={false}>
        {products.map((product: IProduct, key: number) => {
          return (
            <Grid.Column key={product.id + key + "grid"}>
              <SingleProduct product={product} />
            </Grid.Column>
          );
        })}
      </Grid>
    </React.Fragment>
  );
};

export default Products;
