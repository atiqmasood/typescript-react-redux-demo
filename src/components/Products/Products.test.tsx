import * as React from "react";
import Products from "./Products";

import * as renderer from "react-test-renderer";
/**
 * Test suites for products
 */
describe("Products component renders correctly", () => {
  it("Renders Correctly", () => {
    const products = [
      {
        id: 3383096,
        name: "Shirt 'The Perfect Tee Large Batwing'",
        brand: {
          id: 226,
          name: "LEVI'S"
        },
        image: {
          hash: "56755a972fefe61b91380e94df8481e8"
        },
        min_price: 1990,
        max_price: 1990,
        quantity_per_pack: "some quantity"
      }
    ];
    const rendered = renderer.create(<Products products={products} />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
