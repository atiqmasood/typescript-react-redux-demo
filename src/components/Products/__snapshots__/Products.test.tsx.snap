// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products component renders correctly Renders Correctly 1`] = `
<div
  className="ui four column grid"
>
  <div
    className="column"
  >
    <div
      className="ui card"
      href={undefined}
      onClick={[Function]}
    >
      <img
        className="ui image"
        src="//cdn.aboutstatic.com/file/56755a972fefe61b91380e94df8481e8?quality=90&progressive=1&bg=f2f2f2&brightness=0.95&width=400&height=400"
      />
      <div
        className="content"
      >
        <div
          className="header"
        >
          Shirt 'The Perfect Tee Large Batwing'
        </div>
        <div
          className="meta"
        >
          Price : 
          1990
           - 
          1990
        </div>
        <div
          className="description"
        >
          LEVI'S
        </div>
      </div>
      <div
        className="extra content"
      >
        Quantity: 
        some quantity
      </div>
    </div>
  </div>
</div>
`;
